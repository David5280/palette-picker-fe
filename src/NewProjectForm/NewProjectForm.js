import React, { Component } from 'react'

export class NewProjectForm extends Component {
  render() {
    return (
      <form>
        <label htmlFor='new-project-name'>Create a new project:</label>
        <div>
          <input 
            type='text' 
            placeholder='New Project Name...'
            id='new-project-name' 
          />
          <button>Save Project</button>
        </div>
      </form>
    )
  }
}

export default NewProjectForm;
