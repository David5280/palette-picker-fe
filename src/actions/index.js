export const storePalette = palette => ({
  type: 'STORE_PALETTE',
  palette
});

export const changeScheme = scheme => ({
  type: 'CHANGE_SCHEME',
  scheme
});

export const changeVariation = variation => ({
  type: 'CHANGE_VARIATION',
  variation
});

export const retrieveProjects = projects => ({
  type: 'RETRIEVE_PROJECTS',
  projects
});