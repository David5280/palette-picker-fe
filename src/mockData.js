export const mockProjects= [
  {
    "id": 1,
    "name": "Artisinal Cheese Shop",
    "created_at": "2019-08-25T21:04:48.793Z",
    "updated_at": "2019-08-25T21:04:48.793Z"
  },
  {
    "id": 3,
    "name": "Butcher Shop",
    "created_at": "2019-08-25T21:04:48.798Z",
    "updated_at": "2019-08-25T21:04:48.798Z"
  },
  {
    "id": 2,
    "name": "Barber",
    "created_at": "2019-08-25T21:04:48.798Z",
    "updated_at": "2019-08-25T21:04:48.798Z"
  }
]

export const singleMockProject = {
  "id": 1,
  "name": "Artisinal Cheese Shop",
  "created_at": "2019-08-25T21:04:48.793Z",
  "updated_at": "2019-08-25T21:04:48.793Z",
  "palette": [
    {
      "id": 1,
      "name": "Warm",
      "created_at": "2019-08-25T21:04:48.835Z",
      "updated_at": "2019-08-25T21:04:48.835Z",
      "color_1": "#000123",
      "color_2": "#900123",
      "color_3": "#J00123",
      "color_4": "#000123",
      "color_5": "#000123",
      "project_id": 1
    },
    {
      "id": 2,
      "name": "Cool",
      "created_at": "2019-08-25T21:04:48.836Z",
      "updated_at": "2019-08-25T21:04:48.836Z",
      "color_1": "#H00123",
      "color_2": "#A00123",
      "color_3": "#G00123",
      "color_4": "#670123",
      "color_5": "#000123",
      "project_id": 1
    }
  ]
}

export const mockPalettes = [
  {
    "id": 1,
    "name": "Warm",
    "color_1": "#000123",
    "color_2": "#900123",
    "color_3": "#J00123",
    "color_4": "#000123",
    "color_5": "#000123",
    "project_id": 1,
    "created_at": "2019-08-25T21:04:48.835Z",
    "updated_at": "2019-08-25T21:04:48.835Z"
  },
  {
    "id": 3,
    "name": "Oblique",
    "color_1": "#600123",
    "color_2": "#000123",
    "color_3": "#120123",
    "color_4": "#006123",
    "color_5": "#430123",
    "project_id": 3,
    "created_at": "2019-08-25T21:04:48.837Z",
    "updated_at": "2019-08-25T21:04:48.837Z"
  },
  {
    "id": 2,
    "name": "Cool",
    "color_1": "#H00123",
    "color_2": "#A00123",
    "color_3": "#G00123",
    "color_4": "#670123",
    "color_5": "#000123",
    "project_id": 1,
    "created_at": "2019-08-25T21:04:48.836Z",
    "updated_at": "2019-08-25T21:04:48.836Z"
  },
  {
    "id": 4,
    "name": "Winter",
    "color_1": "#000123",
    "color_2": "#000123",
    "color_3": "#120123",
    "color_4": "#005123",
    "color_5": "#430123",
    "project_id": 3,
    "created_at": "2019-08-25T21:04:48.840Z",
    "updated_at": "2019-08-25T21:04:48.840Z"
  },
  {
    "id": 5,
    "name": "Autumn",
    "color_1": "#010123",
    "color_2": "#043123",
    "color_3": "#430123",
    "color_4": "#000123",
    "color_5": "#300123",
    "project_id": 2,
    "created_at": "2019-08-25T21:04:48.841Z",
    "updated_at": "2019-08-25T21:04:48.841Z"
  },
  {
    "id": 6,
    "name": "Rustic",
    "color_1": "#500123",
    "color_2": "#000123",
    "color_3": "#120123",
    "color_4": "#800123",
    "color_5": "#430123",
    "project_id": 2,
    "created_at": "2019-08-25T21:04:48.842Z",
    "updated_at": "2019-08-25T21:04:48.842Z"
  },
  {
    "id": 7,
    "name": "Resonant",
    "color_1": "#600123",
    "color_2": "#060123",
    "color_3": "#120123",
    "color_4": "#003123",
    "color_5": "#431123",
    "project_id": 2,
    "created_at": "2019-08-25T21:04:48.843Z",
    "updated_at": "2019-08-25T21:04:48.843Z"
  }
]

export const singleMockPalette = {
  "id": 1,
  "name": "Warm",
  "color_1": "#000123",
  "color_2": "#900123",
  "color_3": "#J00123",
  "color_4": "#000123",
  "color_5": "#000123",
  "project_id": 1,
  "created_at": "2019-08-25T21:04:48.835Z",
  "updated_at": "2019-08-25T21:04:48.835Z"
}

export const mockColors = [
  '#fffg11',
  '#faf00g',
  '#fffg33',
  '#gsgf90',
  '#fhhggg',
]