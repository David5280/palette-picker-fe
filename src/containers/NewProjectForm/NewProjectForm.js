import React, { Component } from 'react'
import { serverCall } from '../../fetchCalls/fetchCalls';
import { addProject } from '../../actions';
import { connect } from 'react-redux';

export class NewProjectForm extends Component {
  state = {
    projectName: ''
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  createProject = async (e) => {
    e.preventDefault();
    const newProjectRes = await serverCall('projects', 'POST', { name: this.state.projectName })
    this.props.addProject({ name: await newProjectRes.name, id: await newProjectRes.id })
    this.props.submit()
  }

  render() {
    return (
      <form className='project-name-input'>
        <div>
          <input 
            type='text' 
            placeholder='New Project Name...'
            data-test="project-name-test-input"
            value={this.state.projectName}
            onChange={this.handleChange}
            name='projectName'
            className='buttons-left project-name-input'
          />
          <button onClick={(e) => this.createProject(e)} className='buttons-left project-name-input'>Save Project</button>
        </div>
      </form>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  addProject: project => dispatch(addProject(project))
});

export default connect(null, mapDispatchToProps)(NewProjectForm);
