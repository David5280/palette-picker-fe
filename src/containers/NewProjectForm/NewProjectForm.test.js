import React from 'react';
import ReactDOM from 'react-dom';
import { NewProjectForm, mapDispatchToProps } from './NewProjectForm';
import { shallow } from 'enzyme'
import { singleMockProject } from '../../mockData';
import { addProject } from '../../actions';

describe('NewProjectForm Container', () => {
  let wrapper;
  let instance;
  beforeEach(() => {
    wrapper =shallow(
      <NewProjectForm 
        addProject={jest.fn()}
      />
    )
    instance = wrapper.instance()
  })

  it('should match the snap shot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should map dispatch to props', () => {
    const mockDispatch = jest.fn();
    const mockResults = singleMockProject;
    const actionToDispatch = addProject(mockResults);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.addProject(mockResults);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  })


})