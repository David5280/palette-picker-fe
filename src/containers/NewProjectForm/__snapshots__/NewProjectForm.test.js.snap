// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProjectForm Container should match the snap shot 1`] = `ShallowWrapper {}`;
